javasetter/javasetter.go

```diff
@@ -104,18 +102,19 @@ func (j JavaSetter) setJavaMac(version JavaVersion) (Result, error) {
 }
 
 func (j JavaSetter) setJavaUbuntu(version JavaVersion) (Result, error) {
-	javaPath, javaCPath, javadocPath, javaHome := func() (string, string, string, string) {
-		switch version {
-		case JavaVersion8:
-			mainDir := "/usr/lib/jvm/java-8-openjdk-amd64"
-			return mainDir+"/jre/bin/java", mainDir+"/bin/javac", mainDir+"/bin/javadoc", mainDir
-		case JavaVersion11:
-			mainDir := "/usr/lib/jvm/java-11-openjdk-amd64"
-			return mainDir+"/bin/java", mainDir+"/bin/javac", mainDir+"/bin/javadoc", mainDir
-		default:
-			return "", "", "", ""
-		}
-	}()
+	var javaHome string
```

Comment:
This whole code block was a lot more complex than necessary. The key here is that JDK8 had a unique `java` path, but I double-checked it, and `$JAVA_HOME/bin/java`is also present in the JDK8 folder (it's a symlink to `$JAVA_HOME/jre/bin/java`). This means that the relative paths can be shared between all 3 cases, the only unique variable is the JAVA_HOME.